docs/design-document.md

```diff
@@ -572,62 +548,78 @@ The workflow is as follows:
+
+When an actor pushes a change to a remote repository, this update to the
+corresponding ref (or refs) must be recorded in the RSL. For each ref being
+pushed, the gittuf client creates a new RSL entry. Then, `RSLPush` is used to
+submit these changes to the remote repository.
+
+### Force Pushes
+
+Due to the linear nature of the RSL, it is not possible to remove a reference
+entry. A force push makes one or more prior reference entries for the pushed ref
+invalid as the targets recorded in those entries may not be reachable any
+longer. Thus, these entries must be marked as "skipped" in the RSL using an
+annotation entry. After an annotation for these reference entries is created, a
+reference entry is created recording the current state of the ref. Then,
+`RSLPush` is used to submit these changes to the remote repository.
+
+### Verification Workflow
 
 There are several aspects to verification. First, the right policy state must be
 identified by walking back RSL entries to find the last change to that
 namespace. Next, authorized keys must be identified to verify that commit or RSL
 entry signatures are valid.
 
-### Identifying Authorized Signers for Protected Namespaces
+#### Identifying Authorized Signers for Protected Namespaces
 
 When verifying a commit or RSL entry, the first step is identifying the set of
 keys authorized to sign a commit or RSL entry in their respective namespaces.
-With commits, the relevant namespaces pertain to the files they modify, tracked
-by the repository. On the other hand, RSL entries pertain to Git refs. Assume
-the relevant policy state entry is `P` and the namespace being checked is `N`.
-Then:
-
-1. Validate `P`'s Root metadata using the TUF workflow, ignore expiration date
-   checks.
-1. Begin traversing the delegations graph rooted at the top level Targets
-   metadata. Set `current` to top level Targets and `parent` to Root metadata.
-1. Create empty set `K` to record keys authorized to sign for `N`.
-1. While `K` is empty:
-   1. Load and verify signatures of `current` using keys provided in `parent`.
-      Abort if signature verification fails.
-   1. Identify delegation entry that matches `N`, `D`.
-   1. If `D` is the `allow-rule`:
-      1. Explicitly indicate any key is authorized to sign changes as `N` is not
-         protected. Returning empty `K` alone is not sufficient.
-   1. Else:
-      1. If repository contains metadata with the role name in `D`:
-         1. Set `parent` to `current`, `current` to delegatee role.
-         1. Continue to next iteration.
-      1. Else:
-         1. Set `K` to keys authorized in the delegations entry.
+This is achieved by performing pre-ordered depth first search over the
+delegations graph in a gittuf policy state. Assume the relevant policy state
+entry is `P` and the namespace being checked is `N`. Then:
+
+1. Validate `P`'s root metadata using the TUF workflow starting from the initial
+   root of trust metadata, ignore expiration date checks.
```

Comment:
I guess this isn't the place, but the rationale for ignoring this should be explained.

Reply:
#280 I'll add a pointer to this